### PR TITLE
Backport dependencies to supported versions

### PR DESCRIPTION
**Pull request by ying-jeanne**
#### What this PR does
Add Backport Label for Dependency Updates

We have enabled Dependabot for Mimir, but it currently only operates on the main branch. However, in order to keep other supported versions up-to-date with potential security vulnerabilities (CVEs), we need to backport the necessary dependency updates to those branches.

This pull request aims to automatically add a backport label when Dependabot raises pull requests for updating dependencies. By adding the backport label, we can streamline the process of updating dependencies across multiple branches and ensure consistent security across all supported versions.

The backport label will trigger the necessary workflows to automatically apply the dependency updates to the respective branches, allowing us to maintain a secure and up-to-date codebase.
#### Which issue(s) this PR fixes or relates to

Fixes #<issue number>

#### Checklist

- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: .github/workflows/supported_versions.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    "versions": [
+        {
+            "version": "release-2.9",
+            "support_reason": "Standard support schedule",
+            "support_end_date": "2023-12-31"
+        },
+        {
+            "version": "release-2.8",
+            "support_reason": "Standard support schedule",
+            "support_end_date": "2023-12-31"
```
Comment (ying-jeanne) [inline .github/workflows/supported_versions.json:11]: The support end date needed to be double checked.

File: .github/workflows/dependabot_reviewer.yml [hunk 2]
```diff
@@ -23,6 +23,24 @@ jobs:
...
           github-token: "${{ secrets.GITHUB_TOKEN }}"
+      # For the current supported version, update dependencies according to docs
+      - name: Get list of supported version and backport dependencies
+        id: supported-versions
+        run: |
+          supportedVersions=$(jq -r '.versions[] | select(.support_end_date > (now | strftime("%Y-%m-%d"))) | .branch | "backport \(.)"' supported_versions.json | paste -sd, -)
+          
+          # Join array elements with commas
+          joinedVersions=$(printf "%s, " "${supportedVersions[@]}")
+          joinedVersions=${joinedVersions%, }  # Remove the trailing comma
```
Comment (charleskorn) [inline .github/workflows/dependabot_reviewer.yml:34]: I think we can use `jq`'s `join()` function to simplify this: https://jqlang.github.io/jq/manual/#:~:text=Example-,join(str),-Joins%20the%20array

**Review by charleskorn — COMMENTED** [review]
LGTM, just one suggestion to simplify the Bash magic to create a comma-separated list of tags

**Comment by ying-jeanne** [comment]
I will close the PR at the moment, since the backport would not cover all cases (older branch could have different dependencies)

**Pull request closed without merging**